[global]
  sendAnonymousUsage = false

[log]
  level = "DEBUG" # ERROR|INFO
  filePath = "log/traefik.log" # see 'traefik: volumes:' section in the compose file

[providers.docker]
  exposedByDefault = false
  swarmMode = true
  endpoint = "tcp://127.0.0.1:2377" # :2375
  network = "traefik_proxy"
  watch = true # watch for new (or no longer available) containers
  # endpoint = "unix:///var/run/docker.sock"
  # docker swarm mode (1.12+): https://docs.traefik.io/routing/providers/docker/

[api]
  dashboard = true
  insecure = true
  # https://docs.traefik.io/operations/dashboard/#insecure-mode

# https://docs.traefik.io/routing/entrypoints/#configuration
[entryPoints]
  [entryPoints.web]
    address = ":80"
    # [entryPoints.web.redirect]
    #  entryPoint = "web-secure"
  [entryPoints.backend]
    address = ":8080"
  [entryPoints.drone]
    address = ":8088"
  [entryPoints.dashboard]
    address = ":9000"
  [entryPoints.web-secure]
    address = ":443"
    [certificatesResolvers.mycert.acme]
      email = "your-email@your-domain.org"
      storage = "/etc/traefik/acme/acme.json"
      [certificatesResolvers.mycert.acme.httpChallenge]
        entryPoint = "web" # used during the challenge

# TODO: don't forget to do:
# cd config/acme
# chmod 600 acme.json

# https://docs.traefik.io/https/acme/
# https://docs.traefik.io/migration/v1-to-v2/